common-http/src/test/java/io/cdap/common/http/HttpRequestsTestBase.java

```diff
@@ -110,13 +127,39 @@ public void testHttpStatus() throws Exception {
     testDelete("/api/testDelete", only(200), any(), any(), any());
   }
 
+  private HttpRequest getRequest(HttpRequest.Builder builder) {
```

Comment:
I would just pass a `CompletableFuture<String>` to this method and call `future.complete` in the onFinish method. This avoid usage of two separate fields.

Reply:
done